docs/docs/protocol-specs/public-vm/execution.md

```diff
@@ -115,11 +115,13 @@ machineState.daGasLeft = 0
 ### Gas cost notes and examples
 
 An instruction's gas cost is meant to reflect the computational cost of generating a proof of its correct execution. For some instructions, this computational cost changes based on inputs. Here are some examples and important notes:
-- [`JUMP`](./instruction-set/#isa-section-jump) is an example of an instruction with constant gas cost. Regardless of its inputs, the instruction always incurs the same `l1GasCost`, `l2GasCost`, and `daGasCost`.
+
+- All instructions have a base cost. [`JUMP`](./instruction-set/#isa-section-jump) is an example of an instruction with constant gas cost. Regardless of its inputs, the instruction always incurs the same `l1GasCost`, `l2GasCost`, and `daGasCost`.
 - The [`SET`](./instruction-set/#isa-section-set) instruction operates on a different sized constant (based on its `dstTag`). Therefore, this instruction's gas cost increases with the size of its input.
-- Instructions that operate on a data range of a specified "size" scale in cost with that size. An example of this is the [`CALLDATACOPY`](./instruction-set/#isa-section-calldatacopy) argument which copies `copySize` words from `environment.calldata` to `machineState.memory`.
+- In addition to the base cost, the cost of an instruction increases with the number of reads and writes to memory. This is affected by the total number of input and outputs: the gas cost for [`AND`](./instruction-set/#isa-section-and) should be greater than that of [`NOT`](./instruction-set/#isa-section-not) since it takes one more input.
+- Input parameters flagged as "indirect" as they require an extra memory access, so these should further increase the gas cost of the instruction.
```

Comment:
I think remove "as they":
```suggestion
- Input parameters flagged as "indirect" require an extra memory access, so these should further increase the gas cost of the instruction.
```

Reply:
Fixed in https://github.com/AztecProtocol/aztec-packages/commit/dfb87049a92e3e4db72484b527d35326c17c07e9